src/codegen/array_boundary.rs

```diff
@@ -0,0 +1,58 @@
+use super::vartable::Vartable;
+use crate::codegen::cfg::{ControlFlowGraph, Instr};
+use crate::codegen::Expression;
+use crate::sema::ast::Type;
+use indexmap::IndexMap;
+use num_bigint::BigInt;
+use num_traits::One;
+use solang_parser::pt::Loc;
+
+// IndexMap <ArrayVariable res , (res of temp variable, expression of temp variable)>
+pub type ArrayTempVars = IndexMap<usize, (usize, Expression)>;
+
+// Function to modify arraye length temp
+pub(crate) fn modify_temp_array_size(
```

Comment:
As your data structure to track the array length is an attribute of ControlFlowGraph, you can implement this function under `impl ControlFlowGraph`, so that is becomes a method of it.

Reply:
This function just reads from the map, it doesn't modify it

Reply:
Your map is a member of `ControlFlowGraph`. You read from it here and add instructions to a basic block (which is also in `ControlFlowGraph`). As a better design patter, this function can be a member of such data structure.